pkgs/misc/uboot/default.nix

```diff
@@ -34,20 +34,31 @@ let
         sha256 = "0bbw0q027xvzvdxxvpzjajg4rm30a8mb7z74b6ma9q0l7y7bi0c4";
       })
       (fetchpatch {
-        url = https://github.com/dezgeg/u-boot/commit/pythonpath-2017-11.patch;
-        sha256 = "162b2lglp307pzxsf9m7nnmzwxqd7xkwp5j85bm6bg1a38ngpl9v";
+        url = https://github.com/dezgeg/u-boot/commit/pythonpath-2018-03.patch;
+        sha256 = "0bhbnxhpf7x6kk5ikhy6hzf5k132mc0pvin90hn8dbw028d6cyma";
+      })
+      (fetchpatch {
+        url = https://github.com/dezgeg/u-boot/commit/extlinux-path-length-2018-03.patch;
+        sha256 = "07jafdnxvqv8lz256qy29agjc2k1zj5ad4k28r1w5qkhwj4ixmf8";
       })
     ];
 
     postPatch = ''
       patchShebangs tools
     '';
 
-    nativeBuildInputs = [ bc dtc openssl python2 ];
+    nativeBuildInputs = [ bc dtc openssl python2 swig ];
+    depsBuildBuild = [ buildPackages.stdenv.cc ];
 
     hardeningDisable = [ "all" ];
 
-    makeFlags = [ "DTC=dtc" ] ++ extraMakeFlags;
+    makeFlags = [
+      "DTC=dtc"
+      "ARCH=${if hostPlatform.platform.kernelArch == "arm64" then "arm"
```

Comment:
Actually, setting `$ARCH` shouldn't be required with U-Boot?

Reply:
Yeah, it seems to build fine without it.